src/websocket.js

```diff
@@ -0,0 +1,47 @@
+import mitt from 'mitt';
+import { encode } from './encoding';
+
+export default (url, options) => {
+  const emitter = mitt();
+
+  emitter.on('data', message => {
+    // need to access the message data
+    const lines = encode(message);
+
+    emitter.emit('update', { lines });
+  });
+
+  emitter.on('start', () => {
+    try {
+      // try to connect to websocket
+      const socket = new WebSocket(url);
+
+      socket.addEventListener('open', event => {
+        if (options.onOpen) {
+          options.onOpen(event, socket);
+        }
+      });
+      socket.addEventListener('close', options.onClose);
+
+      socket.addEventListener('error', error => {
+        emitter.emit('error', error);
+      });
```

Comment:
I think the error emitter should only be called when there is an error loading the actual url which should get caught in the catch statement.

https://github.com/mozilla-frontend-infra/react-lazylog/blob/de294c475d622221234542bdbc38ffde3f9c9255/src/components/LazyLog/index.jsx#L101-L105

I guess we can just call the onError handler in websocketOptions if it exists. Something like:

```js
const { onError } = options;

// ...

socket.addEventListener('error', error => {
  if (onError) {
    onError(error);
  }
});
```